### PR TITLE
feat: add create metadata key API

### PR DESCRIPTION
**Pull request by ishanarya0**
To support metadata key creation for this [shield issue](https://github.com/odpf/shield/issues/162)

### CODE REVIEW

File: odpf/shield/v1beta1/shield.proto [hunk 1]
```diff
@@ -644,6 +656,24 @@ message CreateUserResponse {
   User user = 1;
 }
 
+message MetadataKeyRequestBody {
+  string key = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
```
Comment (krtkvrm) [inline odpf/shield/v1beta1/shield.proto:660]: How about key only having Upper Caps, like enum?

**Review by krtkvrm — CHANGES_REQUESTED** [review]

File: odpf/shield/v1beta1/shield.proto (hunk 1 above)
Comment (ishanarya0) [reply odpf/shield/v1beta1/shield.proto:660]: This regex shall allow all uppercase keys. 
`ex: STOP, START etc would be valid`

I think it's the desired behavior(?)

Comment (krtkvrm) [reply odpf/shield/v1beta1/shield.proto:660]: It will also allow - `Edit`, `slack-Id`

Comment (ishanarya0) [reply odpf/shield/v1beta1/shield.proto:660]: @krtkvrm Can you please help in listing down the formats/examples of keys that should be accepted and not accepted?

Comment (ravisuhag) [reply odpf/shield/v1beta1/shield.proto:660]: Should we just let me plain string? Any specific need to put validation on this apart from string?

Comment (krtkvrm) [reply odpf/shield/v1beta1/shield.proto:660]: No, just regex that would be acceptable for key
Like `STOP_FIREHOSE` or `Stop-Firehose` or `StopFirehose`.. etc. I would prefer 1st one, open to discussions

Comment (ishanarya0) [reply odpf/shield/v1beta1/shield.proto:660]: I wanted to keep a check on the use of some symbols that generally are not there in keys. Examples of such keys are` @type, stop(), 1==1`. By allowing pain text, users can use anything for the key. 

Shall we still allow plain text @ravisuhag ?

Comment (ishanarya0) [reply odpf/shield/v1beta1/shield.proto:660]: @krtkvrm Yes, shall we allow all of these, including all small caps and snake case?  Ex `manager, manager_email`

Comment (ravisuhag) [reply odpf/shield/v1beta1/shield.proto:660]: +1. I think we can just skip symbols and allow all other characters.

**Review by krtkvrm — APPROVED** [review]

**Pull request merged**